snapcraft/internal/project_loader/_extensions/ros2_dashing.py

```diff
@@ -0,0 +1,73 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018-2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Import types and tell flake8 to ignore the "unused" List.
+
+from typing import Any, Dict, Tuple
+
+from ._extension import Extension
+
+_PLATFORM_SNAP = dict(core18="gnome-3-34-1804")
+
+
+class ExtensionImpl(Extension):
+    """Drives ROS2 build and runtime environment for snap."""
+
+    @staticmethod
+    def get_supported_bases() -> Tuple[str, ...]:
+        # TODO: core20
+        return ("core18",)
+
+    @staticmethod
+    def get_supported_confinement() -> Tuple[str, ...]:
+        return ("strict", "devmode")
+
+    def __init__(self, *, extension_name: str, yaml_data: Dict[str, Any]) -> None:
+        super().__init__(extension_name=extension_name, yaml_data=yaml_data)
+
+        self.root_snippet = {
+            "package-repositories": [
+                {
+                    "type": "apt",
+                    "url": "http://repo.ros2.org/ubuntu/main",
+                    "components": ["main"],
+                    "deb-types": ["deb"],
+                    "key-id": "C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654",
+                    "key-server": "keyserver.ubuntu.com",
+                    "suites": ["$SNAPCRAFT_APT_RELEASE"],
+                }
+            ]
+        }
+
+        self.app_snippet = {
+            "command-chain": ["snap/command-chain/ros2-launch"],
+            "environment": {
+                "ROS_DISTRO": "dashing",
+                "PYTHONPATH": "$SNAP/opt/ros/dashing/lib/python3.6/site-packages:$SNAP/usr/lib/python3/dist-packages:$PYTHONPATH",
+            },
+        }
+
+        self.part_snippet = {"build-environment": [{"ROS_DISTRO": "dashing"}]}
+
+        self.parts = {
+            "ros2-dashing-colcon": {
```

Comment:
This doesn't actually involve colcon, maybe `ros2-dashing-launcher` is a better name. Is there an established pattern for extension parts names?

Reply:
Ah that's a good idea, will do.

Reply:
Done.